src/test/java/com/uid2/admin/vertx/ClientKeyServiceTest.java

```diff
@@ -279,16 +268,46 @@ public void listBySite(Vertx vertx, VertxTestContext testContext) {
         });
     }
 
+    @Test
+    public void setContact(Vertx vertx, VertxTestContext testContext) {
+        fakeAuth(Role.CLIENTKEY_ISSUER);
+
+        setClientKeys(new LegacyClientBuilder().build());
+
+        post(vertx, testContext, "api/client/contact?oldContact=test_contact&newContact=test_contact1", "", response -> {
+            ClientKey expected = new LegacyClientBuilder().withContact("test_contact1").build().toClientKey();
+            assertAll(
+                    "clientSetContact",
+                    () -> assertEquals(200, response.statusCode()),
+                    () -> assertEquals(expected, OBJECT_MAPPER.readValue(response.bodyAsString(), ClientKey.class)),
+                    () -> verify(clientKeyStoreWriter).upload(collectionOfSize(1), isNull())
+            );
+            testContext.completeNow();
+        });
+    }
+
+    @Test
+    public void setContactWithExistingContact(Vertx vertx, VertxTestContext testContext) {
```

Comment:
Is it worth testing the 404 case in a separate test too? `setContactWithUnknownContact`?